File: Python_Folder/Lab_Question/matrix_multiplication.py
r1 = int (input("Enter the no. of row of the first matrix: "))
c1 = int (input("Enter the no. of column of the first matrix: "))
r2 = int (input("Enter the no. of row of the second matrix: "))
c2 = int (input("Enter the no. of column of the second matrix: "))
if c1==r2:
    list1 = []
    list2 = []
    list3 = []
    list = []
    print("Enter elements of first matrix")
    for i in range(0,r1):
        list.clear()
        for j in range(0,c1):
            a = int (input())
            list.append(a)
        list1.append(list)
    print("Enter elements of second matrix")
    for i in range(0,r2):
        list.clear()
        for j in range(0,c2):
            a = int (input())
            list.append(a)
        list2.append(list)
    for i in range(r1):
        list.clear()
        for j in range(c2):
            list.append(0)
        list3.append(list)
    
    for i in range(0,r1):
        list.clear()
        for j in range(0,c2):
            s = 0
            for k in range (0,r2):
                s+=list1[i][k]*list2[k][j]
            list3[i][j] = s
    print("The result matrix is: ")
    for i in range(r1):
        for j in range(c2):
            print(list3[i][j],end = " ")
        print()
else:
    print("Multiplication is not possible.")
    

File: Java_Programmes/Day5/MatrixMultiplication.java
package College_Programmes.Day5;

import java.util.Scanner;

public class MatrixMultiplication {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the number of rows for the first matrix: ");
        int rowsA = scanner.nextInt();
        System.out.print("Enter the number of columns for the first matrix: ");
        int colsA = scanner.nextInt();

        System.out.print("Enter the number of rows for the second matrix: ");
        int rowsB = scanner.nextInt();
        System.out.print("Enter the number of columns for the second matrix: ");
        int colsB = scanner.nextInt();

        // Check if multiplication is possible
        if (colsA != rowsB) {
            System.out.println("Matrix multiplication is not possible.");
        } else {
            // Input the elements of the matrices
            int[][] matrixA = inputMatrix(rowsA, colsA, "first");
            int[][] matrixB = inputMatrix(rowsB, colsB, "second");

            // Multiply the matrices
            int[][] resultMatrix = multiplyMatrices(matrixA, matrixB);

            // Display the result
            System.out.println("\nResultant Matrix (Product of the two matrices):");
            displayMatrix(resultMatrix);
        }
    }

    public static int[][] inputMatrix(int rows, int cols, String matrixName) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter the elements of the " + matrixName + " matrix:");

        int[][] matrix = new int[rows][cols];

        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                System.out.print("Enter element at position (" + (i + 1) + ", " + (j + 1) + "): ");
                matrix[i][j] = scanner.nextInt();
            }
        }
        return matrix;
    }

    public static int[][] multiplyMatrices(int[][] matrixA, int[][] matrixB) {
        int rowsA = matrixA.length;
        int colsA = matrixA[0].length;
        int colsB = matrixB[0].length;

        int[][] resultMatrix = new int[rowsA][colsB];

        for (int i = 0; i < rowsA; i++) {
            for (int j = 0; j < colsB; j++) {
                for (int k = 0; k < colsA; k++) {
                    resultMatrix[i][j] += matrixA[i][k] * matrixB[k][j];
                }
            }
        }

        return resultMatrix;
    }

    public static void displayMatrix(int[][] matrix) {
        for (int[] row : matrix) {
            for (int element : row) {
                System.out.print(element + " ");
            }
            System.out.println();
        }
    }
}

